src/library/impl/meta_reader/table.h

```diff
@@ -334,8 +334,7 @@ namespace xlang::meta::reader
 
         bool operator==(row_base const& other) const noexcept
         {
-            XLANG_ASSERT(m_table == other.m_table);
-            return m_index == other.m_index;
+            return (m_table == other.m_table) && (m_index == other.m_index);
```

Comment:
Why this change? Rows are basically iterators and the assertion says that comparing iterators from different tables is an invalid operation. This is much the same as comparing iterators from different containers in the C++ standard library.

Reply:
when collecting required interfaces, I needed a way to [test if two given types were the same.](https://github.com/Microsoft/xlang/blob/1376a084e300b9f512d939825293199bd96987c4/src/tool/python/helpers.h#L214). If there's a more appropriate way, please let me know.